### DISCUSSION TITLE
Is there a way to get the active instance's context with client API?

### DISCUSSION THREAD

**Discussion by allenscha** [facebook/docusaurus#9157] [Q&A] [answered]
Hello,

Is there an API to get the active instance's context (plugin ID, version, and so on) in a [multi-instance](https://docusaurus.io/docs/docs-multi-instance) setup?

I looked up at the [Docusaurus Client API](https://docusaurus.io/docs/docusaurus-core) page, but it appears that there is no such feature regarding the instance, although there seems to be one for I18n (`currentLocale`).

I'd greatly appreciate your help.

**Comment by slorber** [comment]
We have an undocumented core React hook for that

```tsx
import useRouteContext from '@docusaurus/useRouteContext';

const routeContext = useRouteContext();
routeContext.plugin.name;
routeContext.plugin.id;
```

It's not documented so technically not part of the public API surface yet but I doubt we remove it. Maybe avoid using this in a plugin's code, but if it's your own site that's fine.

**Reply by allenscha** [reply]
Many thanks for the information. I verified that I can get the active plugin's ID using the code you provided, in a custom component for our site.

How about the active version of the instance? It doesn't look like the information is provided by `useRouteContext()`. Could you let me know whether it is also provided in some way or not?

**Reply by slorber** [reply, marked as answer]
This hook is generic/core.

If your React component is only called inside docs pages, you can use these ones:

```tsx
import {
  useVersions,
  useActiveVersion,
  useActivePluginAndVersion,
  useActiveDocContext,
} from '@docusaurus/plugin-content-docs/client';
```

**Reply by allenscha** [reply]
Thank you very much. We're intending to use the React component inside docs pages only, and the hooks above work well for that purpose.
